### PR TITLE
add basic support for macos platform (untested)

### PR DESCRIPTION
**Pull request by sizmailov**
@leberwurstsaft Can you confirm it works on you system?

Should be the final bit of #54

### CODE REVIEW

**Comment by jcwinkler** [comment]
Package containing the fix, for testers:
[TexText-Linux-MacOS-Test.zip](https://github.com/textext/textext/files/2660913/TexText-Linux-MacOS-Test.zip)

**Comment by linneman** [comment]
Hi folks,

I have just check the installation procedure on latest Mac OS X Mojave. It looks quite ok but a minor problem is reported about pstoedit being slightly too new (3.73 instead of 3.70) and the current homebrew version of ghostscript (9.26) not being found at all. The URL's with further instructions which the installers prints out unfortunately do not work. Of course I could compile these packages from the sources but I believe this is not the intention of the package maintainers. So in the hope that this information might be helpful, ere is what I did and what I got:

brew cask install mactex inkscape
brew install pygtk gtksourceview pdf2svg ghostscript pstoedit

python setup.py --inkscape-executable=/usr/local/bin/inkscape

The last command gives the following messages:

[TexText][INFO    ]: Using `inkscape-executable` = `/usr/local/bin/inkscape`
[TexText][SUCCESS ]: + [Succ] TexText requirements
[TexText][SUCCESS ]: /-and-* [Succ] Detect inkscape
[TexText][SUCCESS ]: |      inkscape is found at `/usr/local/bin/inkscape`
[TexText][SUCCESS ]: /-and-* [Succ] Detect `python2.7`
[TexText][SUCCESS ]: |      python2.7 is found at `/usr/local/opt/python@2/bin/python2.7`
[TexText][SUCCESS ]: /-and-+ [Succ] Detect *latex
[TexText][SUCCESS ]: |     /--or-* [Succ] `pdflatex` is found at `/Library/TeX/texbin`
[TexText][SUCCESS ]: |     /--or-* [Succ] `lualatex` is found at `/Library/TeX/texbin`
[TexText][SUCCESS ]: |     /--or-* [Succ] `xelatex` is found at `/Library/TeX/texbin`
[TexText][SUCCESS ]: /-and-+ [Succ] Detect GUI library
[TexText][SUCCESS ]: |     /--or-* [Succ] PyGTK2 is found
[TexText][SUCCESS ]: |     /--or-* [Succ] TkInter is found
[TexText][SUCCESS ]: /-and-+ [Succ] Detect pdf->svg conversion utility
[TexText][INFO    ]:       /--or-+ [Fail] Detect compatible pstoedit+ghostscript versions
[TexText][INFO    ]:       |      Please follow installation instructions at 
[TexText][INFO    ]:       |             https://sizmailov.github.io/textext/install/macos.html#macos-install-pstoedit
[TexText][SUCCESS ]:       |     /-and-+ [Succ] 
[TexText][INFO    ]:       |     |     /-not-+ [Fail] Detect incompatible versions of pstoedit+ghostscript
[TexText][INFO    ]:       |     |           /-and-* [Fail] pstoedit=3.70 is not found (but pstoedit=3.73 is found)
[TexText][INFO    ]:       |     |           /-and-* [Fail] ghostscript=9.22 is not found
[TexText][SUCCESS ]:       |     /-and-* [Succ] pstoedit is found
[TexText][INFO    ]:       |     /-and-* [Fail] ghostscript is not found
[TexText][INFO    ]:       |            Please follow installation instructions at 
[TexText][INFO    ]:       |                   https://sizmailov.github.io/textext/install/macos.html#macos-install-pstoedit
[TexText][INFO    ]:       |            If ghostscript is installed in custom location, specify it via 
[TexText][INFO    ]:       |                   --ghostscript-executable=<path-to-ghostscript>
[TexText][INFO    ]:       |            and run setup.py again
[TexText][SUCCESS ]:       /--or-* [Succ] Detect pdf2svg:
[TexText][SUCCESS ]:              `pdf2svg` is found at `/usr/local/bin`
[TexText][INFO    ]: `textext/config.json` is not found in distribution, keep old file
[TexText][INFO    ]: Removing `/Users/ol/.config/inkscape/extensions/textext`
[TexText][INFO    ]: Removing `/Users/ol/.config/inkscape/extensions/textext.inx`
[TexText][INFO    ]: Copying `extension/textext.inx` to `/Users/ol/.config/inkscape/extensions/textext.inx`
[TexText][INFO    ]: Creating directory `/Users/ol/.config/inkscape/extensions/textext`
[TexText][INFO    ]: Copying `extension/textext/win_app_paths.py` to `/Users/ol/.config/inkscape/extensions/textext/win_app_paths.py`
[TexText][INFO    ]: Copying `extension/textext/utility.pyc` to `/Users/ol/.config/inkscape/extensions/textext/utility.pyc`
[TexText][INFO    ]: Copying `extension/textext/requirements_check.py` to `/Users/ol/.config/inkscape/extensions/textext/requirements_check.py`
[TexText][INFO    ]: Copying `extension/textext/default_packages.tex` to `/Users/ol/.config/inkscape/extensions/textext/default_packages.tex`
[TexText][INFO    ]: Copying `extension/textext/errors.pyc` to `/Users/ol/.config/inkscape/extensions/textext/errors.pyc`
[TexText][INFO    ]: Copying `extension/textext/utility.py` to `/Users/ol/.config/inkscape/extensions/textext/utility.py`
[TexText][INFO    ]: Copying `extension/textext/typesetter.py` to `/Users/ol/.config/inkscape/extensions/textext/typesetter.py`
[TexText][INFO    ]: Copying `extension/textext/__init__.py` to `/Users/ol/.config/inkscape/extensions/textext/__init__.py`
[TexText][INFO    ]: Copying `extension/textext/textext.log` to `/Users/ol/.config/inkscape/extensions/textext/textext.log`
[TexText][INFO    ]: Copying `extension/textext/errors.py` to `/Users/ol/.config/inkscape/extensions/textext/errors.py`
[TexText][INFO    ]: Copying `extension/textext/latexlogparser.py` to `/Users/ol/.config/inkscape/extensions/textext/latexlogparser.py`
[TexText][INFO    ]: Copying `extension/textext/asktext.py` to `/Users/ol/.config/inkscape/extensions/textext/asktext.py`
[TexText][INFO    ]: Copying `extension/textext/requirements_check.pyc` to `/Users/ol/.config/inkscape/extensions/textext/requirements_check.py

Hope this helps, thanks for the support in advance and 
Best Regards,

Otto

**Comment by jcwinkler** [comment]
Thanks @linneman! Are you able to call ghostscript from the command line using the command `ghostscript` ? Or is there another command used for it?

@sizmailov : The links to textext.github.io are already included in the current develop branch. MacOS pages need to be written, of course, depending on our results.

> [TexText][SUCCESS ]: | /-and-+ [Succ]
> [TexText][INFO ]: | | /-not-+ [Fail] Detect incompatible versions of pstoedit+ghostscript
> [TexText][INFO ]: | | /-and-* [Fail] pstoedit=3.70 is not found (but pstoedit=3.73 is found)
> [TexText][INFO ]: | | /-and-* [Fail] ghostscript=9.22 is not found

In fact this is a positive result (no incompatible versions found) but it leads to misinterpretation, as we see. Maybe we find another way to present it so there is no confusion?

**Comment by linneman** [comment]
The command 'ghostscript' is not recognized on my system but gs is:

solaris:~ ol$ gs
GPL Ghostscript 9.26 (2018-11-20)
Copyright (C) 2018 Artifex Software, Inc.  All rights reserved.
This software comes with NO WARRANTY: see the file PUBLIC for details.
GS>quit

**Comment by sizmailov** [comment]
@linneman Thank you for report. TexText supports two options for GUI, three options for *latex, and two pdf-to-svg converters. Another check is made for incompatible versions of psedit+ghostscript. It complicates the output of the requirements check, complaining about missing parts. Docs for macos is not ready yet. 

I agree. Information about missing, but not strictly required parts might be confusing. Some rewording should help with it, e.g. replace Succ/Fail/Undf by T/F/U; "Please follow installation instructions at" -> "To install %missing part% follow instructions at"

@jcwinkler We can resolve the confusion by putting BIG SUCCESS MESSAGE at the end of log along with "Optional dependencies of TexText are not found, but it's ok" (on success only)

Presumably we should add extra section to installation instructions with explanation of the `setup.py` output

File: extension/textext/requirements_check.py [hunk 1]
```diff
@@ -53,6 +53,22 @@ def call_command(command):
...
 
+class MacDefaults(LinuxDefaults):
+    os_name = "macos"
+    executable_names = {"inkscape": ["inkscape-bin"],
+                        "pdflatex": ["pdflatex"],
+                        "lualatex": ["lualatex"],
+                        "xelatex": ["xelatex"],
+                        "pdf2svg": ["pdf2svg"],
+                        "pstoedit": ["pstoedit"],
+                        "ghostscript": ["ghostscript"]}
```
[the remark this answers is not in the document]
Comment (sizmailov) [reply extension/textext/requirements_check.py]: Seems so

**Pull request merged**